packages/react-utils/src/helpers/tests/dataLoaders.test.ts

```diff
@@ -18,11 +20,85 @@ jest.mock('@opensrp/pkg-config', () => {
   };
 });
 
+jest.mock('@opensrp/server-service', () => ({
+  __esModule: true,
+  ...Object.assign({}, jest.requireActual('@opensrp/server-service')),
+}));
+
+describe('helpers/dataLoaders/fetchProtectedImage', () => {
+  (global as any).URL.createObjectURL = jest.fn().mockReturnValue('hello');
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    fetch.resetMocks();
+    jest.restoreAllMocks();
+  });
+
+  beforeAll(() => {
+    store.dispatch(
+      authenticateUser(
+        true,
+        {
+          email: 'bob@example.com',
+          name: 'Bobbie',
+          username: 'RobertBaratheon',
+        },
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        { api_token: 'hunter2', oAuth2Data: { access_token: 'hunter2', state: 'abcde' } }
+      )
+    );
+  });
+
+  it('returns the url of the protected image', async () => {
+    const bytes = new Uint8Array(59);
+
+    for (let i = 0; i < 59; i++) {
+      bytes[i] = 32 + i;
+    }
+
+    fetch.mockResponseOnce(JSON.stringify([bytes.buffer]));
+
+    const objectURL = await fetchProtectedImage(
+      'https://mg-eusm-staging.smartregister.org/opensrp/multimedia/media/4'
+    );
+
+    await flushPromises();
+    expect(fetch.mock.calls).toHaveLength(1);
+    expect(fetch.mock.calls[0]).toEqual([
+      'https://mg-eusm-staging.smartregister.org/opensrp/multimedia/media/4',
+      {
+        headers: {
+          authorization: 'Bearer hunter2',
+        },
+        method: 'GET',
+      },
+    ]);
+    /***
+     * @todo Most appropriate test case would be to assert
+     * expect((global as any).URL.createObjectURL).toHaveBeenCalledWith(blob);
+     * but mocking a blob response was a challenge
+     */
+    expect(objectURL).toEqual('hello');
```

Comment:
@kelvin-muchiri please add a docstring explaining where 'hello' comes from

Reply:
nevermind, I see its mocked at the top

Reply:
@p-netm Ok...I have updated the comment